### ISSUE TITLE
`StyleBoxTexture` does not propagate its `Texture2D`'s `changed` signal

### ISSUE THREAD

**Issue by Hop311** [godotengine/godot#89045]
### Tested versions

- Reproducible in v4.2.1.stable [b09f793f5]

### System information

Godot v4.2.1.stable - Windows 10.0.19045 - Vulkan (Forward+) - dedicated NVIDIA GeForce RTX 4060 Ti (NVIDIA; 31.0.15.4629) - 13th Gen Intel(R) Core(TM) i7-13700KF (24 Threads)

### Issue description

`StyleBoxTexture` does not connect its `Texture2D`'s `changed` signal to its own, so modification to the texture aren't automatically propagated.

### Steps to reproduce

Using the attached project:
- Run the project and click "Update Texture".
- The button on the right won't have changed, mouse over it to force it to redraw and show the change (the button's icon toggling between showing the left and right side).
- Click "Toggle Connection" to connect the texture's `changed` signal to the stylebox's `emit_changed`.
- Now clicking "Update Texture" will immediately change the button on the right.
- Click "Toggle Connection" again to disconnect the `changed` signal.
- Now once again clicking "Update Texture" won't show any change until you mouse over the button on the right.

### Minimal reproduction project (MRP)

[StyleBoxTexture-Changed-Signal-Propagation.zip](https://github.com/godotengine/godot/files/14462203/StyleBoxTexture-Changed-Signal-Propagation.zip)

**Comment by AThousandShips** [comment]
This is true for most resources though, the ones that propagate change are the exception

**Comment by Spartan322** [comment]
From a surface level, all the other inbuilt `StyleBox` propertues are non-reference Variants, `StyleBoxTexture`'s texture property is the only `StyleBox` property that could emit a `changed` signal, IDK how the the Godot team thinks of this, kinda seems like a unique case and it feels like it be more intuitive, least to me, for this case to be part of the exception.

**Comment by AThousandShips** [comment]
More so than `Material`? Or `MultiMesh`? Or `GradientTexture`?

**Comment by Spartan322** [comment]
Sure, but aren't there already exceptions in some cases for this type of behavior?

**Comment by AThousandShips** [comment]
Very few, why should this one be an exception?

**Comment by Spartan322** [comment]
I think it would depend on the justification for the other exceptions before it could be reasoned if this would be a valid concern then.
